### PR TITLE
Make address area searchable and visible within Elasticsearch

### PR DESCRIPTION
**Pull request by vfarah-if**
### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

**Review by vfarah-if — COMMENTED** [review]
Adding more ES edge cases

File: datahub/company/test/factories.py [hunk 1]
```diff
@@ -110,6 +112,18 @@ class Meta:
         model = 'company.Company'
 
 
+class USCompanyFactory(CompanyFactory):
```
Comment (elcct) [inline datahub/company/test/factories.py:115]: For setting up specific fixtures, the convention is to use pytest.fixtures, for example: 
https://github.com/uktrade/data-hub-api/blob/develop/datahub/search/company/test/test_public_search_view.py#L20

File: datahub/core/constants.py [hunk 2]
```diff
@@ -24,6 +33,324 @@ class Country(Enum):
     united_states = Constant('United States', '81756b9a-5d95-e211-a939-e4115bead28a')
 
 
+class AdministrativeArea(Enum):
+    """Administrative Areas"""
+
+    # United States
+    alabama = AdministrativeAreaConstant(
```
Comment (elcct) [inline datahub/core/constants.py:40]: These constants are only used in tests, so I am not sure it is worth adding this many. I'd leave only those that are used in tests.

File: datahub/search/investment/test/test_elasticsearch.py [hunk 3]
```diff
@@ -778,6 +776,12 @@ def test_get_basic_search_query():
         'track_total_hits': True,
     }
 
+    query = get_basic_search_query(
+        ESInvestmentProject, 'test', offset=5, limit=5,
```
Comment (elcct) [inline datahub/search/investment/test/test_elasticsearch.py:780]: It would be good to have arguments in separate lines.

File: datahub/company/test/factories.py (hunk 1 above)
Comment (vfarah-if) [reply datahub/company/test/factories.py]: Fixed

File: datahub/search/investment/test/test_elasticsearch.py (hunk 3 above)
Comment (vfarah-if) [reply datahub/search/investment/test/test_elasticsearch.py]: Fixed

File: changelog/elasticarea.feature.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Address area elastic data changes making address area searchable.
```
Comment (elcct) [inline changelog/elasticarea.feature.md:1]: Maybe this needs rewording to something like "Companies are now searchable by administrative area."

**Review by elcct — APPROVED** [review]

**Review by tomkingston — APPROVED** [review]
Looks good, but I agree with Pawel that the feature description could be reworded to be a bit clearer

File: changelog/elasticarea.feature.md (hunk 4 above)
Comment (vfarah-if) [reply changelog/elasticarea.feature.md:1]: Fixed ...
